src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,25 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.UserRegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user != null) {
+            if (user.getLogin() != null && user.getLogin().length() >= 6) {
+                if (user.getPassword() != null && user.getPassword().length() >= 6) {
+                    if (user.getAge() != null && user.getAge() >= 18) {
+                        if (storageDao.get(user.getLogin()) == null) {
+                            return storageDao.add(user);
+                        }
+                    }
+                }
+            }
+        }
```

Comment:
The nested if-statements make the code less readable. Consider using early returns to simplify the logic. For example, check for null or invalid conditions first and throw the exception immediately, then proceed with the registration logic.